Commands/HelpCommand.cs

```diff
@@ -31,22 +34,52 @@ public async Task ModHelp ()
             await ReplyAsync(string.Empty, false, helpDialog);
         }
 
+        /// <summary>
+        /// Returns the help dialog for a specific mode.
+        /// </summary>
         private EmbedBuilder GetHelpDialog(string mode)
         {
             EmbedBuilder eb = new EmbedBuilder()
                 .WithColor(new Color(247, 22, 131))
                 .WithTitle("BrackeysBot")
                 .WithDescription("The official Brackeys server bot. Commands are:");
 
-            string prefix = BrackeysBot.Configuration["prefix"];
+            string prefix = "[]";
 
-            List<KeyValuePair<string, string>> commands = BrackeysBot.Help.GetCommands(mode);
-            foreach (KeyValuePair<string, string> command in commands)
+            var commands = GetCommandDatas(mode);
+            foreach (var command in commands)
             {
-                eb.AddField(prefix + command.Key, command.Value);
+                string title = prefix + command.Usage;
+                string content = command.Description;
+                eb.AddField(title, content);
             }
 
             return eb;
         }
+
+        /// <summary>
+        /// Gathers the data for all commands found in the command modules.
+        /// </summary>
+        private IEnumerable<HelpDataAttribute> GetCommandDatas (string mode)
+        {
+            var commandList = new List<HelpDataAttribute>();
+
+            foreach (ModuleInfo module in _commands.Modules)
+                foreach (CommandInfo command in module.Commands)
+                {
+                    var data = command.Attributes.FirstOrDefault(a => a is HelpDataAttribute);
+                    if (data != default(HelpDataAttribute))
```

Comment:
Why are you checking if the data (of type `HelpDataAttribute`) has all default values? That data should never have default values, you're already expecting that it has data.

Reply:
My bad for unclear coding. default(HelpDataAttribute) is null, so I am basically checking if the command has a HelpDataAttribute.

Reply:
Then it's simpler and more clear to just check if it's null.